src/v2/conversions.rs

```diff
@@ -914,19 +944,52 @@ impl TryFrom<AccountInfo> for super::types::AccountInfo {
             index,
             stake,
             address,
+            cooldowns,
+            available_balance,
         } = value;
         let account_nonce = sequence_number.require()?.into();
         let account_amount = amount.require()?.into();
-        let account_release_schedule = schedule.require()?.try_into()?;
+        let account_release_schedule: AccountReleaseSchedule = schedule.require()?.try_into()?;
         let account_threshold = threshold.require()?.try_into()?;
         let account_encrypted_amount = encrypted_balance.require()?.try_into()?;
         let account_encryption_key = encryption_key.require()?.try_into()?;
         let account_index = index.require()?.into();
-        let account_stake = match stake {
+        let account_stake: Option<super::types::AccountStakingInfo> = match stake {
             Some(stake) => Some(stake.try_into()?),
             None => None,
         };
         let account_address = address.require()?.try_into()?;
+        let mut cds = Vec::with_capacity(cooldowns.len());
+        for cooldown in cooldowns.into_iter() {
+            cds.push(cooldown.try_into()?)
+        }
+        let cooldowns = cds;
+
+        // The available balance is only provided as convenience and in case the
+        // calculation of it changes in the future. If the available balance is
+        // not present, we calculate it manually instead. Note that changes in the
+        // calculation method may make this calculation invalid.
+        let available_balance = available_balance.map(|ab| ab.into()).unwrap_or_else(|| {
+            // According to the protobuf documentation:
+            // The available (unencrypted) balance of the account is the balance...
+            let balance = account_amount;
+
+            // ... minus the locked amount. The locked amount is the maximum of the
+            // amount in the release schedule...
+            let release_schedule_amount = account_release_schedule.total;
+
+            // ... and the total amount that is actively staked or in cooldown (inactive
+            // stake).
+            let staked_amount = account_stake.as_ref()
+                .map(|s| s.staked_amount())
+                // FIXME: Give common::types::Amount a `Default` impl and use .unwrap_or_default();
+                .unwrap_or(common::types::Amount::zero());
```

Comment:
I'm not really sure about this one? Is this the "total amount that is actively staked or in cooldown (inactive stake)" or is this not correct? I'm wondering if the amounts in the cooldowns need to be summed or something.

Reply:
You seem to do that in your PR for the node but there you have something with precooldown and pre-precooldown that I'm not sure about :thinking:

Reply:
The stake should be the sum of the active stake and all of the amounts in cooldown (i.e. the inactive stake). In the node implementation, the cooldowns are separated into pre-pre-cooldown, pre-cooldown and cooldown proper in the data model, but in what the API returns, it is just a list of all of them, with tags distinguishing.

Reply:
Okay, hopefully the implementation is correct now :)